src/Pages/UserPages/Explore/Explore.tsx

```diff
@@ -1,3 +1,262 @@
+}
 export default function Explore() {
-  return <div>Explore</div>;
+  const location: PickerData = useLocation();
+  console.log(location);
+  const [favoriteRooms, setFavoriteRooms] = useState<FavoriType[]>([]);
+  const [rooms, SetRooms] = useState<roomType[]>(
+    location.state?.data?.rooms || []
+  );
+  const [totalCount, setTotalCount] = useState<number>(
+    location.state?.data?.totalCount || 1
+  );
+  const [page,setPage] = useState<number>(1);
+  const [size,setSize] = useState<number>(8);
+
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [selectedFilter, setSelectedFilter] = useState<
+    "highest" | "lowest" | null
+  >(null);
+
+  const getAllRooms = async (page:number , size:number) => {
+    try {
+      const response = await apiClient.get<AllRoomsResponseType>(
+        getRoomDetails,{
+          params:{page:page , size: size}
+        }
+      );
+
+      SetRooms(response.data.data.rooms);
+      setTotalCount(response.data.data.totalCount);
+    } catch (error) {
+      const axiosError = error as AxiosError<{ message: string }>;
+      toast.error(
+        axiosError.response?.data?.message || "An unexpected error occurred."
+      );
+    }
+  };
+
+  const getMyFavoriRooms = async () => {
+    try {
+      const response = await apiClient.get<FavoriResponseType>(
+        PORTAL_URLS.favoriRoom
+      );
+
+      setFavoriteRooms(response.data.data.favoriteRooms);
+    } catch (error) {
+      const axiosError = error as AxiosError<{ message: string }>;
+      toast.error(
+        axiosError.response?.data?.message || "An unexpected error occurred."
+      );
+    }
+  };
+  const handleFavoriteToggle = async (
+    roomId: string,
+    action: "add" | "remove"
+  ) => {
+    const toastId = toast.loading("Processing...");
+    setIsLoading(true);
+    try {
+      const response =
+        action === "add"
+          ? await apiClient.post(PORTAL_URLS.favoriRoom, {
+              roomId,
+            })
+          : await apiClient.delete(`${PORTAL_URLS.favoriRoom}/${roomId}`, {
+              data: { roomId },
+            });
+      getMyFavoriRooms();
+      toast.success(response.data.message, {
+        id: toastId,
+      });
+      setIsLoading(false);
+    } catch (error) {
+      const axiosError = error as AxiosError<{ message: string }>;
+      toast.error(
+        axiosError.response?.data?.message || "An unexpected error occurred.",
+        {
+          id: toastId,
+        }
+      );
+      setIsLoading(false);
+    }
+  };
+
+  const checkIfRoomInFavori = (id: string) => {
+    return favoriteRooms.some((room) =>
+      room?.rooms.some((room: roomType) => room._id === id)
+    );
+  };
+  const sortRooms = (value: "highest" | "lowest") => {
+    setSelectedFilter(value);
+    SetRooms(
+      rooms.sort((a, b) =>
+        value === "highest" ? b.price - a.price : a.price - b.price
+      )
+    );
+  };
+  useEffect(() => {
+    if (!location.state?.data?.rooms) {
```

Comment:
instead of location state we can use searchParams instead when navigating
```
navigate("explore?startDate=`${startDate}&endDate=${endDate}`")
```

and then get it in the explore page using the searchParams with the key startDate and endDate